### PR TITLE
Drop Newtonsoft

### LINKED ISSUE #243: Resolve Newtonsoft warning
`The version of Newtonsoft.Json cannot be greater than 9.0.1, but version 12.0.3 is currently being used.	`

### PR DESCRIPTION
**Pull request by TimHess**
Resolve #243 by switching to `System.Text.Json`

### CODE REVIEW

**Review by atwoosnam — DISMISSED** [review]

**Review by zzori-theoriginal — DISMISSED** [review]

**Review by atwoosnam — DISMISSED** [review]

**Comment by TimHess** [comment]
Surprisingly, this may not be a complete solution... I'm still seeing the warnings on the build server, and did just get it to show up locally again (though it doesn't always show up right away)

File: src/VisualStudioExtension/test/Tanzu.Toolkit.VisualStudioExtension.Tests/Tanzu.Toolkit.VisualStudioExtension.Tests.csproj [hunk 1]
```diff
@@ -15,6 +15,7 @@
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
+    <PackageReference Include="Newtonsoft.Json" Version="13.0.1" />
```
Comment (atwoosnam) [inline src/VisualStudioExtension/test/Tanzu.Toolkit.VisualStudioExtension.Tests/Tanzu.Toolkit.VisualStudioExtension.Tests.csproj:18]: Huh, so is Newtonsoft still used in the VS project then?

Comment (atwoosnam) [reply src/VisualStudioExtension/test/Tanzu.Toolkit.VisualStudioExtension.Tests/Tanzu.Toolkit.VisualStudioExtension.Tests.csproj:18]: Oh I have a vague memory of Newtonsoft being a dependency of another package (maybe Community VS Toolkit?) that was specific to the package proj iirc... maybe that's what's going on?

**Review by atwoosnam — APPROVED** [review]

File: src/VisualStudioExtension/test/Tanzu.Toolkit.VisualStudioExtension.Tests/Tanzu.Toolkit.VisualStudioExtension.Tests.csproj (hunk 1 above)
Comment (TimHess) [reply src/VisualStudioExtension/test/Tanzu.Toolkit.VisualStudioExtension.Tests/Tanzu.Toolkit.VisualStudioExtension.Tests.csproj:18]: Pretty sure `MSTest.TestAdapter` has a ref, which conflicts with one from somewhere else. Long story short, this ref is only to resolve the conflict

**Pull request merged**